### PR TITLE
Remove Karma + Mocha + Chai + Sinon setup

### PR DESCRIPTION
**Pull request by greypants**
Just use [Jest](https://facebook.github.io/jest/)! 

It used to be super complicated to string the right series of tools together to get a cohesive and full featured JS test suite — which is why we previously did it for you. But now Jest exists, solves these issues and is our strong recommendation.

This cuts out a TON of dependencies, and significantly lightens up the package.

Also removes some old README files that have since been consolidated.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,67 +14,55 @@
...
   },
   "scripts": {
     "start": "gulp",
     "gulp": "gulp",
     "build": "gulp build",
     "deploy": "gulp deploy",
-    "test": "karma start --single-run",
-    "test:watch": "karma start",
-    "test:gulp": "mocha ./gulpfile.js/**/*.test.js"
+    "dev:test": "mocha ./gulpfile.js/**/*.test.js"
```
Comment (mackermedia) [inline package.json]: why not just `yarn test` ?

Comment (greypants) [inline package.json]: this one's tests for the package itself, the others are part of the public API (sort of)

Comment (mackermedia) [inline package.json]: ahh okay. got it.

**Comment by mackermedia** [comment]
👍

File: README.md [hunk 2]
```diff
@@ -6,7 +6,6 @@
 
 ## Quick start on a fresh project (empty directory)
 ```bash
-yarn init
```
Comment (greypants) [inline README.md]: this line's not needed... if you `yarn add` and a `package.json` and `yarn.lock` file don't exist, one will be created.

**Pull request merged**